### ISSUE TITLE
There hasn't been a change to openProcess. Yet, as you can see, you are missing a handle. Which is probably also the reason why openProcess fails

### ISSUE THREAD

**Issue by NyshMy** [qb-0/pyMeow#59]
open_process(process: string|int,  debug: bool = false): Process  
But just need to pass a process PID or process name, I can open it normally with 32-bit programs, some x64 programs don't work, and I use CE to see if the process has permission to be accessed

**Comment by NyshMy** [comment]
Sorry, I clicked the wrong button and opened a new question,
However, when I use ctypes to call the OpenProcess (0x1ffff, False, pid) API to open the same process, I can open the handle normally, which proves that the permissions are sufficient. In other words, I can also open the process using the previous pymeow. When I downloaded the latest pymeow, I found that there was no open process, so it was very strange

**Comment by NyshMy** [comment]
def __init__(self):
        self.pid = pyMeow.get_process_id('pycharm64.exe')
        self.Process = pyMeow.open_process(self.pid, debug=True)
        print(f"Process:{self.Process}")
Process:{'name': 'pycharm64.exe', 'pid': 83428, 'debug': True, 'base': 5368709120, 'handle': 464}

    for proc in pyMeow.enum_processes():
        print(proc)
{'name': 'DNF.exe', 'pid': 20852, 'debug': False, 'base': 0, 'handle': 0}
{'name': 'SGuardSvc64.exe', 'pid': 83512, 'debug': False, 'base': 0, 'handle': 0}

nimpy.Exception: Unexpected error encountered: Module 'DNF.exe' not found
At present, that's all I can provide. I'm not sure if this exception is caused by the process not being found or by the OpenProcess API

**Comment by NyshMy** [comment]
I have tested the get'module function with the previous version of PyMeow and it works fine. It can return the module address, but some methods may not work because I used the previous version. However, in the latest version, I cannot open the Process properly, so I cannot test the get'module method on the new PyMeow

**Comment by NyshMy** [comment]
I think I may not be able to compile. I used to download packaged pyd files, so can I just wait for qb's approval

**Comment by NyshMy** [comment]
Hint: used config file 'F:\nim-2.0.8\config\nim.cfg' [Conf]
Hint: used config file 'F:\nim-2.0.8\config\config.nims' [Conf]
........................................................................
Error: cannot open 'F:\nim-2.0.8\bin\pyMeow.nim'

**Comment by NyshMy** [comment]
I pulled down the latest code and recompiled pymeow, but still get the error self. Process=pyMeow. open_decess (self. pid, debug=True)
nimpy.Exception: Unexpected error encountered: Module 'DNF.exe' not found， Can I only use the old version of Pymeow now

**Comment by NyshMy** [comment]
def __init__(self):
        self.pid = pyMeow.get_process_id('DNF.exe')
        print('process_pid:', self.pid)
        self.Process = pyMeow.open_process(self.pid, debug=True)

process_pid: 79944
Traceback (most recent call last):
  File "F:/Myproject/.vscode/x64rw.py", line 103, in <module>
    process = Memory()
  File "F:/Myproject/.vscode/x64rw.py", line 15, in __init__
    self.Process = pyMeow.open_process(self.pid, debug=True)
nimpy.Exception: Unexpected error encountered: Module 'DNF.exe' not found

**Comment by NyshMy** [comment]
@Hypnootika 

proc getModule(process: Process, moduleName: string): Module {.exportpy: "get_module".} =
for module in enumModules(process):
echo module
if moduleName == module.name:
return module
echo "end found"
raise newException(ValueError, fmt"Module '{moduleName}' not found")
I made changes to the code locally and found that the echo module did not print anything. Instead, it threw an exception saying 'Module' {moduleName} 'not found before printing' end found '. Therefore, I suspect that there was an error in the for module in enum Modules (process): this error was caused by the program not going into the loop, and even when I pulled the latest code, it would still look like this

**Comment by NyshMy** [comment]
@Hypnootika 
File "F:/Myproject/.vscode/pymeow_test/x64rw.py", line 35, in r_int
    return pyMeow.r_int(self.Process, address)
nimpy.AccessViolationDefect: Unexpected error encountered: Read failed [Address: 0x000000014B9CE0A0] [Error: 5 - 拒绝访问。]
Process:{'name': 'DNF.exe', 'pid': 20392, 'debug': True, 'handle': 424}


I used the code in the fork (branch "patch-1"), and indeed did not use getModule. However, I seem to have encountered a new problem where I am encountering exceptions when applying for memory or retrieving module addresses. Please take a look again. Thank you very much, as I am not very familiar with the nim language, some questions may appear slightly inferior

**Comment by NyshMy** [comment]
444/10000
实时翻译
划译
It is normal for me to use these methods with a 32-bit process, but using my DNF.exe will return an empty list
self.pid = pyMeow.get_process_id('DNF.exe')
print('process_pid:', self.pid)
self.Process = pyMeow.open_process(self.pid, debug=True)
# self.allow = pyMeow.allocate_memory(self.Process, 1024)
# print(self.allow)
# pyMeow.free_memory(self.Process, 3120168960)
self.module = [x for x in pyM

**Comment by NyshMy** [comment]
It is normal for me to use these methods with a 32-bit process, but using my DNF.exe will return an empty list
    def __init__(self):
        self.pid = pyMeow.get_process_id('DNF.exe')
        print('process_pid:', self.pid)
        self.Process = pyMeow.open_process(self.pid, debug=True)
        self.module = [x for x in pyMeow.enum_modules(self.Process)]
        print(self.module)

**Comment by NyshMy** [comment]
It is normal for me to use these methods with a 32-bit process, but using my DNF.exe will return an empty list
def init(self):
    self.pid = pyMeow.get_process_id('DNF.exe')
    print('process_pid:', self.pid)
    self.Process = pyMeow.open_process(self.pid, debug=True)
    self.module = [x for x in pyMeow.enum_modules(self.Process)]
    print(self.module)

**Comment by NyshMy** [comment]
I'm very sorry, I can use the pycharm64.exe process. I don't know what the problem is with my process, so I'm sorry to trouble you. Thank you very much. I'll just use the previous pymeow version

**Comment by NyshMy** [comment]
@Hypnootika 
I'm glad I solved the problem because my process doesn't have read and write permissions. I loaded a driver to make it read and write, and it worked fine. Thank you very much. It seems that the branch version is missing a few functions
get_proc_address(moduleName,  functionName: string): uint
create_remote_thread(process: Process,  startAddress: uint, param: uint): bool
inject_library(process: Process,  dllPath: string): bool
By the way, I also want to ask a question. Can nim implement x64 calling conventions? Because some games' calls may require
